wazo_confd/plugins/user/middleware.py

```diff
@@ -93,6 +95,51 @@ def create(self, body, tenant_uuid, tenant_uuids):
 
         return user_dict
 
-    def delete(self, user_id, tenant_uuids):
+    def delete(self, user_id, tenant_uuids, recursive=False):
         user = self._service.get(user_id, tenant_uuids=tenant_uuids)
-        self._service.delete(user)
+        if not recursive:
+            self._service.delete(user)
+        else:
+            if user.groups:
+                # dissociation
+                self._middleware_handle.get(
+                    'user_group_association'
+                ).associate_all_groups({'groups': []}, user.uuid)
+
+            for line in user.lines:
+                self._middleware_handle.get('user_line_association').dissociate(
+                    user.uuid, line.id, tenant_uuids
+                )
+                self._middleware_handle.get('line').delete(
+                    line.id, tenant_uuids, recursive=True
+                )
+                Session.expire(user, ['user_lines'])
+
+            for incall in user.incalls:
+                for extension in incall.extensions:
+                    self._middleware_handle.get(
+                        'incall_extension_association'
+                    ).dissociate(incall.id, extension.id, tenant_uuids)
+                    self._middleware_handle.get('extension').delete(
+                        extension.id, tenant_uuids
+                    )
+                self._middleware_handle.get('incall').delete(incall.id, tenant_uuids)
+
+            for switchboard in user.switchboards:
+                members = []
+                for user_member in switchboard.user_members:
+                    if user_member.uuid != user.uuid:
+                        members.append({'uuid': user_member.uuid})
+                self._middleware_handle.get('switchboard_member').associate(
+                    {'users': members}, switchboard.uuid, tenant_uuids
+                )
+            Session.expire(user, ['switchboard_member_users'])
+
+            self._service.delete(user)
+
+            try:
+                self._wazo_user_service.get(user.uuid)
```

Comment:
Is the `get` required before deleting?

Reply:
yes, otherwise a deletion (on a non-existing auth user) raises a 500 ☹️

Reply:
I created https://github.com/wazo-platform/wazo-test-helpers/pull/80 and remove this `get`